### DISCUSSION TITLE
Mathematics of Population Growth

### DISCUSSION THREAD

**Discussion by josePereiro** [josePereiro/TankeFactory#8] [Discussions]
This is just an annoying issue I'm having with a solved problem... I trust the answer but I lack an intuitive interpretation of it...

## Problem formulation

Let's say we have a population of self replicative agents. The replication process is so that an old agent (the mother) is destroyed and replace by two new independent agents (the children). Each agent can be characterized by a duplication time $\tau$ (the time it will spend alive before duplicating). Any new agent duplication time is sampled from a population distribution $p(\tau)$ so this distribution remains constant in time. 

There are no limits on resources or space, so the population duplication continues forever and exponentially. That is, the time evolution of the population (number of agents) is well described by $N(t) = N(0)\exp({\tilde{k}t})$, where $\tilde{k}$ is the population duplication rate (number of times the number of agents gets doubled per unit of time). Note that there is a simple relationship $\tilde{k} = \log{2}/\tilde{\tau}$ between the duplication rate and the duplication time.

> IMPORTANT: We will use the tilde for differentiate population parameters from single agent parameters (e.g. $\tilde\tau \not\equiv \tau$).

Constraints summary:
- The population has always the same duplication time distribution $p(\tau)$
- The population is growing exponentially with duplication rate $\tilde{k}$

## Question

The aim is to compute the population duplication rate $\tilde{k}$ from the agent duplication time distribution $p(\tau)$. This can be done by implicitly solving this integral [[1](https://doi.org/10.1146/annurev.mi.22.100168.002511)]: 

$\int_0^{\infty} \exp(-\tilde{k} \tau) p(\tau) d\tau = 1/2$  

> Note that for the simple case where $p(\tau) = \delta{(\tau = \tau_0)}$ the expression $\tilde{k} = \log{2}/\tau_0$ is recovered

I'm having problems understanding what this expression is telling us about the process. Why it is the consequence of the constraints of the problem. Surprisingly this is a very tricky problem to grasp given that it appear to be straightforward "given the agents constant duplication rate distribution, what is duplication rate of the whole population?". 

Any intuition are welcome.

**Comment by ta-rekt** [comment]
I'm thinking about this by the way. It's a very nice exercice.

**Reply by ta-rekt** [reply]
Okay I think I got it. After trying for three days to prove it, I think it's just a matter of definition, there is nothing to prove. 

Think about the random process of the number of offspring of a single agent $i$, call it $N_i(t)$, where $N_i(0) = 1$ and $N_i(\tau) = 2$, and it increments by $1$ every time a new offspring is born from the lineage of $i$.
What is the expected number of offspring of $i$ at time $t$? It's

$$\mathbb{E}[N_i(t)] = \lim_{N(0) \to \infty}\frac{1}{N(0)}\sum_{i=1}^{N(0)}N_i(t) \approx \frac{N(t)}{N(0)} = \exp(\tilde{k}t)$$

So if you start with one agent, on average you can treat it at time $t$ as $\exp(\tilde{k}t)$ agents. But forget about agents and $\tilde{k}$. Imagine you have a portfolio of initially one dollar, and the interest (or inflation) rate is $\lambda$. Its value at time $t$, in the limit of small compounding intervals, becomes $\exp(\lambda t)$ if it is interest, and $\exp(-\lambda t)$ if it is inflation. Suppose you want to cash out your money after a while according to an exit strategy, which is random and distributed according to $p(\tau)$. What's the expected value of your money at the time you cash out? It's the weighed average of the interest-adjusted, or inflation-adjusted value of your portfolio at the time of cashing out. This is

$$ \int_0^{\infty} \exp(a\lambda \tau) p(\tau) d\tau $$

with $a$ being $-1$ or $1$ depending on if it's inflation or interest.
This quantity is a function of $\lambda$. In fact if $a = -1$ it's the Laplace transform of $p(\tau)$. Now we can _define_ a $\lambda$ such that this is exactly $2$ (for interest), which means on average we double our money, or $1/2$ (for inflation), which means we lose half. If the integral converges, these two values of $\lambda$ will be the same, and this is $\tilde{k}$.

Back to agents, that integral is just saying that the doubling rate is defined as the rate at which the expected (or total, same thing, as we saw above) number of agents has doubled (or halved) by the time it produces an offspring. You could define a tripling rate the same way, or if you had a death process, a replacement rate such that the population stays constant on average. I just looked it up, Cuba has a fertility rate of 1.71, which is below the replacement rate of 2.1. I am more than happy to offer my assistance :}

Let's do an example. Suppose a (hombre) agent reproduces after encountering a bunch of chicas. He always tries two, then reproduces with the third, and suppose he meets them after random exponentially distributed intervals. So $\tau$ is just the sum of three independent exponential r.v.s, say with rate $\alpha$. The Laplace transform of one exponential is $\alpha/(\lambda + \alpha)$, and the PDF of a sum of I.I.D r.v.s is a convolution with itself 3 times, which in Laplace space means taking the cube, so

$$ L_p(\lambda) =  \frac{\alpha^3}{(\alpha+\lambda)^3}$$

Setting this to be $1/2$, we get $\tilde{\lambda} = \alpha(2^{1/3} - 1)$. As a sanity check, if he reproduces with the first chica he meets, his personal doubling rate which is $\alpha$,  and the doubling rate of the population should be equal. This is the case, since the exponent of $2$ will be $1$.

**Reply by josePereiro** [reply]
I'm struggling from the beginning. I don't understand: 

$$\mathbb{E}[N_i(t)] = \lim_{N(0) \to \infty}\frac{1}{N(0)}\sum_{i=1}^{N(0)}N_i(t) \approx \frac{N(t)}{N(0)} = \exp(\tilde{k}t)$$

A naive interpretation of $\mathbb{E}[N_i(t)]$ for me would be:

$$\mathbb{E}[N_i(t)] = \lim_{n \to \infty} \frac{1}{n} \sum_{j}^n N_i^j (t)$$

where the sum runs across $n$ realizations of the random process ($j$ is just an index here). 

I can't reconcile both definitions. 
- Why are you using $N(0)$ as the upper limit of the sum?
- Why are you summing across the agents $i$?
- How is it possible that $N(0) \to \infty$ and $\mathbb{E}[N_i(t)]$ is finite?

Im digesting the rest...

**Comment by ta-rekt** [comment]
I thought about this a bit more. A cleaner answer to your question, what this integral is saying about the process: it's saying that a population of agents that obey $p$ behaves like a population of "average" agents, whose dividing time has no memory, i.e. is exponentially distributed. This is the link that gives you a natural way to define the duplication time/rate for an arbitrary $p$.

If there was such a memoryless agent, $p(\tau) = \lambda\exp(-\lambda\tau)$, the integral would become 

$$\int_0^{\infty} \lambda\exp(-2\lambda\tau)d\tau = \lambda \big\[ -\frac{1}{2\lambda}\exp(-2\lambda\tau) \big\]_0^{\infty} = \frac{1}{2}$$

Notice that there is an assumption here, which is that *if all the agents are the same*, they should be growing at the same rate $\lambda$ as the entire population. This is trivial and happens automatically for the memoryless agent, but observing that it should also be true for any $p$ is the crucial bit. Since for a general $p$ there is no "rate", the integral from your exercise is the natural way to define it.

So you can read it as saying: "define the doubling rate $\tilde{k}$ for an arbitrary $p$ such that in expectation, an agent dividing according to $p$ will duplicate at the same time as the entire population, which itself is identical to a memoryless agent doubling at rate $\tilde{k}$"
